File: QubitServer/src/main/java/org/labrad/qubits/channeldata/TriggerDataBase.scala
package org.labrad.qubits.channeldata

import org.labrad.qubits.channels.TriggerChannel

abstract class TriggerDataBase extends TriggerData {

  private var channel: TriggerChannel = null

  def setChannel(channel: TriggerChannel): Unit = {
    this.channel = channel
  }

  def getChannel(): TriggerChannel = {
    channel
  }
}

File: QubitServer/src/main/java/org/labrad/qubits/channeldata/TriggerDataBase.java
package org.labrad.qubits.channeldata;

import org.labrad.qubits.channels.TriggerChannel;

public abstract class TriggerDataBase implements TriggerData {

  private TriggerChannel channel;

  public void setChannel(TriggerChannel channel) {
    this.channel = channel;
  }

  public TriggerChannel getChannel() {
    return channel;
  }
}
